docs/config/cluster.rst

```diff
@@ -1319,10 +1318,11 @@ The following settings can be used to configure the behavior of the
 
   The ``gateway.recover_after_data_nodes`` setting defines the number of data
   nodes that need to be started before the cluster state recovery will start.
-  Ideally the value of the setting should be equal to the number of data nodes
-  in the cluster, because you only want the cluster state to be recovered once
-  all data nodes are started. However, the value must be bigger than the half
-  of the expected number of data nodes in the cluster.
+  The value must be bigger than half the expected
+  number of data nodes in the cluster.
+  Only set this to a number below the total number of data nodes
+  after checking the number of nodes required for quorum and the number of 
```

Comment:
I don't understand your example, how can the quorum be reached with 2 nodes started out of 100?

Reply:
You can have a quorum with 1 node in a 100 node cluster, with 1 node set to `node.master=true` and all others to `node.master=false`.

Only master nodes are relevant for a quorum. -> a setting that only checks for data nodes is completely irrelevant for any quorum.

Reply:
I agree that this setting is not related to any quorum, but also not to any master node settings (that's why I was confused by your example).
I suggest that we keep the original description and just remove the "value must be bigger than the half of the expected number of data nodes" constrain.
```
The ``gateway.recover_after_data_nodes`` setting defines the number of data nodes
that need to have joined the cluster before the local shard recovery will start.
Ideally the value of the setting should be equal to the number of data nodes
in the cluster, because you only want the cluster state to be recovered once
all data nodes are started to avoid that shards are re-allocated
(replicas promoted to primary, new replicas are allocated).
```

Reply:
> Ideally the value of the setting should be equal to the number of data nodes

I slightly disagree with this statement. This is the role of `expected_data_nodes`. `recover_after_data_nodes` is a lower bound to still recover, if the later is not met. Maybe we can avoid recommendation - as they alway depend on the concrete setup - and rather focus on explaining what the setting does. E.g. in a multi-zone setup the argument for replicas is irrelevant, if zone awareness is enforced.

> you only want the cluster state to be recovered once all data nodes are started

Would also be careful with this, as data recovery is independent of cluster state recovery. We had big confusions around this topics (i.e. that cluster availability is somewhat independent from data availability)

Reply:
@proddata Ah yes right, I was in the wrong setting context. Like you says, `recover_after_data_node` is the minimum threshold to wait with a timeout for the desired final state of `expected_data_nodes`. So +1 to remove any recommendation.

> Would also be careful with this, as data recovery is independent of cluster state recovery.

Also totally agreeing on this, it isn't about the cluster state recovery but shard recovery (data).
Let's remove any recommendation.